src/main/java/org/openlmis/integration/dhis2/service/communication/ProcessedDataExchangeService.java

```diff
@@ -54,10 +60,13 @@ public class ProcessedDataExchangeService {
   @Autowired
   private SharedFacilityRepository sharedFacilityRepository;
 
+  @Autowired
+  private PeriodMappingRepository periodMappingRepository;
+
   /**
    * Sends data from OpenLMIS to DHIS2.
    */
-  public void sendData(Schedule schedule) {
+  public void sendData(Schedule schedule, UUID periodMappingId, List<String> facilityCodes) {
```

Comment:
this method is diffucult to read and it's too big, can you split this to smaller parts ?

Reply:
I created some private methods. 
It is recommended to refine the code in the future and move minor functionalities to separate components.